app/src/common/shared/org/mozilla/vrbrowser/ui/keyboards/JapaneseKeyboard.java

```diff
@@ -0,0 +1,210 @@
+        if (StringUtils.isEmpty(aComposingText)) {
+            mComposingText.clear();
+            return null;
+        }
+
+        // Autocomplete when special characters are clicked
+        char lastChar = aComposingText.charAt(aComposingText.length() - 1);
+        boolean autocompose = mAutocompleteEndings.indexOf(lastChar) >= 0;
+
+        aComposingText = aComposingText.replaceAll("\\s","");
+        if (aComposingText.isEmpty()) {
+            return null;
+        }
+
+        initializeComposingText(aComposingText);
+
+        List<Words> words = new ArrayList<>();
+        int candidates = mConverter.predict(mComposingText, 0, -1);
+        if (candidates > 0) {
+            WnnWord word;
+            while ((word = mConverter.getNextCandidate()) != null) {
+                words.add(new Words(1, word.stroke, word.candidate));
+            }
+        }
+
+        CandidatesResult result = new CandidatesResult();
+        result.words = words;
+
+        if (autocompose) {
+            result.action = CandidatesResult.Action.AUTO_COMPOSE;
+            result.composing = aComposingText;
+
+            mComposingText.clear();
+
+        } else {
+            result.action = CandidatesResult.Action.SHOW_CANDIDATES;
+            result.composing = mComposingText.toString(ComposingText.LAYER2);
+        }
+
+        return result;
+    }
+
+    @Override
+    public CandidatesResult getEmojiCandidates(String aComposingText) {
+        ComposingText text = new ComposingText();
+        mSymbolsConverter.convert(text);
+
+        List<Words> words = new ArrayList<>();
+        int candidates = mSymbolsConverter.predict(mComposingText, 0, -1);
+        if (candidates > 0) {
+            WnnWord word;
+            while ((word = mSymbolsConverter.getNextCandidate()) != null) {
+                words.add(new Words(1, word.stroke, word.candidate));
+            }
+        }
+
+        CandidatesResult result = new CandidatesResult();
+        result.words = words;
+        result.action = CandidatesResult.Action.SHOW_CANDIDATES;
+        result.composing = aComposingText;
+
+        return result;
+    }
+
+    @Override
+    public String getComposingText(String aComposing, String aCode) {
+        return "";
+    }
+
+    private void initializeComposingText(String text) {
+        mComposingText.clear();
+        for (int i=0; i<text.length(); i++) {
+            mComposingText.insertStrSegment(ComposingText.LAYER0, ComposingText.LAYER1, new StrSegment(text.substring(i, i+1)));
+            mPreConverter.convert(mComposingText);
+        }
+        mComposingText.debugout();
+    }
+
+    @Override
+    public boolean supportsAutoCompletion() {
+        return true;
+    }
+
+    @Override
+    public boolean usesComposingText() {
+        return true;
+    }
+
+    @Override
+    public String getKeyboardTitle() {
+        return StringUtils.getStringByLocale(mContext, R.string.settings_language_traditional_chinese, getLocale());
```

Comment:
This should be `settings_language_japanese`. The keyboard title shown in the language list is wrong because of this.

Reply:
@MortimerGoro fixed